modules/selection/main/index.coffee

```diff
@@ -16,9 +16,41 @@ getMethod = (node, methodName) ->
   else
     Element.prototype[methodName]
 
+getMatches = (node) ->
+  # XXX: use hx.vendor
+  matchesMethod =
+    if node.matches then 'matches'
+    else if node.matchesSelector then 'matchesSelector'
+    else if node.mozMatchesSelector then 'mozMatchesSelector'
+    else if node.webkitMatchesSelector then 'webkitMatchesSelector'
+    else if node.oMatchesSelector then 'oMatchesSelector'
+    else if node.msMatchesSelector then 'msMatchesSelector'
+
+  if matchesMethod then getMethod(node, matchesMethod)
+  else (selector) ->
+    matchingNodes = getMethod(this, 'querySelector').call(this, selector)
+    i = matchingNodes.length
+    i while --i >= 0 and matchingNodes.item(i) isnt this
```

Comment:
* What is the type of matchingNodes here ? is it a pseudo-array ?
* Why do we start at the end and work backward ?
* Can we
```
this in Array::slice.call matchingNodes
```
?